### ISSUE TITLE
Streaming for Parallel tool execution

### ISSUE THREAD

**Issue by MayankShah1** [instructor-ai/instructor#922]
**Is your feature request related to a problem? Please describe.**
I am interested in streaming responses for two parallel tool calls. 

**Describe the solution you'd like**
I am trying to develop a comparison tool for summary information across two entities. The information is retrieved from data for the two entities of interest in parallel and then should be streamed. Then the final comparison tool across these information retrieved should be streamed. The streaming is done to ensure minimum latency on the UI.

**Describe alternatives you've considered**
I have tried leveraging the code for Parallel tools - 

`
from __future__ import annotations

import openai
import instructor

from typing import Iterable, Literal
from pydantic import BaseModel
from openai import OpenAI,AzureOpenAI

class Weather(BaseModel):
    location: str
    units: Literal["imperial", "metric"]

class GoogleSearch(BaseModel):
    query: str

client = AzureOpenAI(
    azure_endpoint=<API_ENDPOINT>
    api_key=<API_KEY>,
    api_version=<API_VERSION>,
)

client = instructor.from_openai(
    client, mode=instructor.Mode.PARALLEL_TOOLS
)  

function_calls = client.chat.completions.create(
    model=<DEPLOYMENT_NAME>,
    messages=[
        {"role": "system", "content": "You must always use tools"},
        {
            "role": "user",
            "content": "What is the weather in toronto and dallas and who won the super bowl?",
        },
    ],
    response_model=Iterable[Weather | GoogleSearch],  
    stream=True
)

for fc in function_calls:
    print(fc)
    #> location='Toronto' units='metric'
    #> location='Dallas' units='imperial'
    #> query='super bowl winner'
`

**Additional context**
Error : AssertionError: stream=True is not supported when using PARALLEL_TOOLS mode

**Comment by jxnl** [comment]
not a priority, would be open to a PR

**Comment by jxnl** [comment]
Closing as part of repository maintenance for issues created before 2025.
